src/gui/addnewtorrentdialog.cpp

```diff
@@ -175,6 +176,20 @@ void AddNewTorrentDialog::setTopLevel(bool value)
     SettingsStorage::instance()->storeValue(KEY_TOPLEVEL, value);
 }
 
+int AddNewTorrentDialog::savePathHistoryLength()
+{
+    const int defaultHistoryLength = 8;
+    bool ok = false;
+    const int res = settings()->loadValue(KEY_SAVEPATHHISTORYLENGTH, defaultHistoryLength).toInt(&ok);
+    return ok ? res : defaultHistoryLength;
```

Comment:
This method should return valid value. But config file can contain the invalid one for some reason. So we need normalize it.

Reply:
I tried to manually change value in qbt config and it doesn't matter, when i set to some char or text, default value will be used ( in this case 8 `defaultHistoryLength = 8;` ), when I change higher than is max. range value in the spinner ( it is 100 now ), eg I set to 230, so in this case will be set to 100, the same with values under 0, when I set to -10, spinner value will be set to 0.

Reply:
Done.